kuma/search/paginator.py

```diff
@@ -37,6 +38,18 @@ def page(self, number):
         assigns the count from the ES result to the Paginator.
         """
         number = self.validate_number(number)
+        if number >= 1000:
+            # Anything >=1,000 will result in a hard error in
+            # Elasticsearch which would happen before we even get a chance
+            # to validate that the range is too big. The error you would
+            # get from Elasticsearch 6.x is something like this:
+            #
+            #     Result window is too large, from + size must be less
+            #     than or equal to: [10000] but was [11000].
+            #
+            # See https://github.com/mdn/kuma/issues/6092
+            raise InvalidPage('Page number too large')
```

Comment:
Nit: I would say a better spot for this would be within the `validate_number` method above:

```python
    def validate_number(self, number):
        """
        Validates the given 1-based page number.
        This class overrides the default behavior and ignores the upper bound.
        """
        try:
            number = int(number)
        except (TypeError, ValueError):
            raise PageNotAnInteger('That page number is not an integer')
        if number < 1:
            raise EmptyPage('That page number is less than 1')
        if number >= 1000:
            # Anything >=1,000 will result in a hard error in
            # Elasticsearch which would happen before we even get a chance
            # to validate that the range is too big. The error you would
            # get from Elasticsearch 6.x is something like this:
            #
            #     Result window is too large, from + size must be less
            #     than or equal to: [10000] but was [11000].
            #
            # See https://github.com/mdn/kuma/issues/6092
            raise InvalidPage('Page number too large')
        return number
```

Reply:
Got it!